Sming/Components/Network/src/Network/TcpConnection.cpp

```diff
@@ -423,6 +423,29 @@ err_t TcpConnection::internalOnConnected(err_t err)
 	return res;
 }
 
+bool TcpConnection::enableSsl(const String& hostName)
+{
+	if(tcp == nullptr) {
+		return false;
+	}
+
+	if(tcp->state != ESTABLISHED) {
+		return false;
+	}
+
+	if(!sslCreateSession()) {
+		return false;
+	}
+
+	if(hostName) {
+		ssl->hostName = hostName;
+	}
```

Comment:
Is check necessary? Or is that in case there's an `sslInit` callback registered which might do it. Maybe document to clarify? Or Just assign `ssl->hostName = hostName`